### PR TITLE
Little refactor

### PR DESCRIPTION
**Pull request by hdamico**
## What does this PR do?

Organizes the code and gets rid of some validations

### CODE REVIEW

File: app/helpers/project_builder.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module ProjectBuilder
```
Comment (santib) [inline app/helpers/project_builder.rb:1]: My personal experience is that these techniques using modules to share behavior is hard to maintain/extend. They can quickly grow out of control, start having dependencies between modules and bidirectional access with the classes that includes this modules. Also it's hard to follow the code since you can call methods without being explicit on where that method is defined. My recommendation is just using other POROs if you want to encapsulate behavior

File: app/models/event.rb [hunk 2]
```diff
@@ -27,19 +27,4 @@ class Event < ApplicationRecord
   belongs_to :project
 
   validates :name, :data, presence: true
-  validates :handleable, presence: true, if: proc { |event| event.handled_type? }
-  validates_with EventNameValidator, if: proc { |event| handleable && event.handled_type? }
-  validates :occurred_at, presence: true, if: proc { |event| event.handled_type? }
```
Comment (santib) [inline app/models/event.rb:32]: why are we removing validations?

File: app/helpers/project_builder.rb (hunk 1 above)
Comment (santib) [inline app/helpers/project_builder.rb:1]: Moreover the `helpers` folder in Rails has a meaning and is to use methods directly in the `views`. Which BTW is pattern I think should be use with caution too.

File: app/models/event.rb (hunk 2 above)
Comment (hdamico) [reply app/models/event.rb:32]: The first validation is in the github service already, we are duplicating code and making the validation twice.
The second one I believe is not necessary at all (correct me if I'm wrong)
The third one, the same as the first one.

Comment (santib) [reply app/models/event.rb:32]: @mrubi-rootstrap might know better. Regarding the first and third validation, I get your point but remember that if in the future there is another piece of code creating/updating events then we'll be missing these validations

Comment (hdamico) [reply app/models/event.rb:32]: So would it be better to validate it only in the model? I would like to keep those only in one place.

Comment (santib) [reply app/models/event.rb:32]: ideally we would do business validations outside models and integrity validations in the models. What do you think these 2 are business or integrity validations?

File: app/services/github_service.rb [hunk 3]
```diff
@@ -13,41 +13,37 @@ def call
...
-    Event.create!(project: @project,
-                  data: @payload,
-                  name: @event,
-                  handleable: @entity,
-                  occurred_at: occurence_time(event_name: @event, payload: @payload))
+    project = Projects::Builder.fetch_or_create(@payload['repository'])
+    event = Event.new(project: project, data: @payload, name: @event)
+    if handleable_event?
+      @entity = find_or_create_event_type
+      event.tap do |evnt|
```
Comment (santib) [inline app/services/github_service.rb:25]: I think you don't need to use `tap` here

File: app/services/github_service.rb [hunk 4]
```diff
@@ -13,41 +13,37 @@ def call
...
+        evnt.assign_attributes(handleable: @entity, occurred_at: occurence_time)
+        evnt.save!
+      end
+    else
+      event.tap(&:save!)
```
Comment (santib) [inline app/services/github_service.rb:30]: same here

File: app/models/event.rb (hunk 2 above)
Comment (hdamico) [reply app/models/event.rb:32]: I believe is most likely to be a business validation rather than an integrity validation.

File: app/services/github_service.rb (hunk 3 above)
Comment (hdamico) [reply app/services/github_service.rb:25]: I need to return the `event` here, and `.save` does not return the object, instead it returns `true` or `false`.

Comment (santib) [reply app/services/github_service.rb:25]: you still have event, this would be the same
```
  event.assign_attributes(handleable: @entity, occurred_at: occurence_time)
  event.save!
  event
```

Comment (hdamico) [reply app/services/github_service.rb:25]: Alright

File: config/rails_best_practices.yml [hunk 5]
```diff
@@ -28,7 +28,8 @@ RemoveUnusedMethodsInHelpersCheck: { except_methods: [
   'GithubEventPayloadHelper#occurence_time',
```
Comment (santiagovidal) [inline config/rails_best_practices.yml:28]: Can you remove this and any other method exceptions that are no longer valid?

Comment (hdamico) [reply config/rails_best_practices.yml:28]: Yep!
